core/controllers/profile.py

```diff
@@ -251,7 +251,8 @@ class SignupPage(base.BaseHandler):
     def get(self):
         """Handles GET requests."""
         return_url = str(self.request.get('return_url', self.request.uri))
-
+        if re.match('^/[^//]', return_url) is None:
```

Comment:
I don't think I understand this pattern. Could you explain it?

Also, missing backend tests.

Reply:
Hi,

`^/[^//]` This pattern will only accept those URLs which starts from '/' and doesn't starts with  '//'
Example : '/hello' will be accepted and '//hello' will be rejected.

I'm rejecting the URLs starting with '//' because they are the bypass for URL Redirection 
`oppia.org/signup?return_url=//hello.com` will redirect to `hello.com` &
`oppia.org/signup?return_url=/hello.com` will redirect to `oppia.org/hello.com`

Also , all the URLs starting with https or http will also be rejected and if no match is found with the above pattern the site will be redirectd to '/' which is the learnerdashboard.

I feel its a good practice to redirect to internal directories '/something' than redirecting to https | http URLs need you views on the same :)

I'll run the backend tests.

Thanks

Reply:
> I'll run the backend tests.

Just to be clear, I meant **writing** backend tests to ensure that your fix achieves the behaviour you mention.

Also, redirecting without https/http SGTM. Thanks!

Reply:
I'll write the backend tests and update you on this with the converge reports, hope to meet the desired results Thanks !

Reply:
Hi @seanlip ,

I've a doubt in writing tests for this, i'm matching the current host address with the redirected host address to find that the redirection is in same host address.or not.
will it be a good approach to write a testcase for the above problem?

Reply:
I think you can just check that different return URLs that exemplify the different cases go to the right places.

Reply:
Yes, This can also be done.